File: src/main/java/pro/sky/java/weatherapp/domain/Weather.java
package pro.sky.java.weatherapp.domain;

import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class Weather {
    private String name;
    private WeatherMain main;
    private WeatherDescription weather;

    public void setWeather(WeatherDescription[] weatherDescriptions) {
        weather = weatherDescriptions[0];
    }
}

File: src/main/kotlin/pro/sky/java/weatherapp/domain/Weather.kt
package pro.sky.java.weatherapp.domain

class Weather {
    var name: String? = null
    var main: WeatherMain? = null
    var weather: List<WeatherDescription>? = null

    override fun toString(): String {
        return "Weather(name=$name, main=$main, weather=$weather)"
    }
}